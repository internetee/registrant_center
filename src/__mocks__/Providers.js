import { CookiesProvider } from 'react-cookie';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import translations from '../translations';
import state from './mockReduxState';

const lang = 'et';

export default function Providers({ children }) {
    return (
        <BrowserRouter>
            <Provider store={state}>
                <CookiesProvider>
                    <IntlProvider
                        key={lang}
                        defaultLocale="et"
                        locale={lang}
                        messages={translations[lang]}
                    >
                        {children}
                    </IntlProvider>
                </CookiesProvider>
            </Provider>
        </BrowserRouter>
    );
}
