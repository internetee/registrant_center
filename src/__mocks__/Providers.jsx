import { CookiesProvider } from 'react-cookie';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router';
import translations from '../translations';
import PropTypes from 'prop-types';

const lang = 'et';

export default function Providers({ children, store }) {
    return (
        <BrowserRouter>
            <Provider store={store}>
                <CookiesProvider>
                    <IntlProvider
                        defaultLocale="et"
                        key={lang}
                        locale={lang}
                        messages={translations[lang]}
                    >
                        {children}
                    </IntlProvider>
                </CookiesProvider>
            </Provider>
        </BrowserRouter>
    );
}

Providers.propTypes = {
    children: PropTypes.node.isRequired,
    store: PropTypes.object.isRequired,
};
