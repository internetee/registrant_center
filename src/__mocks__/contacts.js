const contacts = [
    {
        id: "cfbfbb76-aed8-497a-91c1-48d82cbc4588",
        name: "Test Registrant",
        code: "TEST:REGISTRANT",
        ident: {
            code: "39988776655",
            type: "priv",
            country_code: "EE",
        },
        email: "test@registrant.ee",
        phone: "+372.1234567",
        fax: null,
        address: {
            street: null,
            zip: null,
            city: null,
            state: null,
            country_code: null,
        },
        auth_info: "16c57aa7c435b959179c7b",
        statuses: ["ok", "linked"],
        disclosed_attributes: [],
    },
    {
        id: "700829af-4bdd-4c5f-8389-f6568e2ba4ad",
        name: "Test Tech",
        code: "TEST:TECH",
        ident: {
            code: "12345678910",
            type: "priv",
            country_code: "EE",
        },
        email: "test@tech.ee",
        phone: "+372.1234567",
        fax: null,
        address: {
            street: null,
            zip: null,
            city: null,
            state: null,
            country_code: null,
        },
        auth_info: "32776bf4c1d6995e6aa657",
        statuses: ["ok", "linked"],
        disclosed_attributes: ["email", "name"],
    },
    {
        id: "528240a3-3f9e-4d9a-83a2-3b3a43cf0dc7",
        name: "Test Admin",
        code: "TEST:ADMIN",
        ident: {
            code: "10987654321",
            type: "priv",
            country_code: "EE",
        },
        email: "test@admin.ee",
        phone: "+372.1234567",
        fax: null,
        address: {
            street: null,
            zip: null,
            city: null,
            state: null,
            country_code: null,
        },
        auth_info: "0e81a4b9049eda45c823a9",
        statuses: ["ok", "linked"],
        disclosed_attributes: ["email"],
    },
]

export default contacts
