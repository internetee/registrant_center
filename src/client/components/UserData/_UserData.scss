.user-data {
  padding: 20px;
  text-align: center;
  @include gradient-gray(right);
  h2 {
    @include h2();
    margin-bottom: 40px;
  }
  h4 {
    @include h4();
  }
  &--item {
    i {
      display: block;
      margin: 0 auto 20px;
      font-size: 50px;
      line-height: 50px;
    }
  }
  @include min-screen(768px) {
    padding: 40px 20px 140px;
    margin-bottom: -40px;
  }
  @include min-screen(1224px) {
    padding: 40px 50px 160px;
  }
}