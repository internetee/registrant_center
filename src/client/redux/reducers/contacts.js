import api from '../../utils/api';
import {
  FETCH_CONTACT_REQUEST,
  FETCH_CONTACT_SUCCESS,
  FETCH_CONTACT_FAILURE,
  FETCH_CONTACTS_REQUEST,
  FETCH_CONTACTS_SUCCESS,
  FETCH_CONTACTS_FAILURE,
  LOGOUT_USER
} from '../actions';

const request = {
  data: [],
  offset: 0,
};

const requestContacts = () => {
  return {
    type: FETCH_CONTACTS_REQUEST,
    isLoading: true
  };
};

const requestContact = () => {
  return {
    type: FETCH_CONTACT_REQUEST,
    isLoading: true
  };
};

const receiveContact = (data, state) => {
  return {
    type: FETCH_CONTACT_SUCCESS,
    status: 200,
    data: state.contacts.data.map(item => (item.id === data.id) ? {...item, ...data} : item),
    isLoading: false,
    isInvalidated: false,
  };
};

const receiveContacts = (data) => {
  return {
    type: FETCH_CONTACT_SUCCESS,
    status: 200,
    data,
    isLoading: false,
    isInvalidated: false,
  };
};


const receiveAllContacts = (data) => {
  return {
    type: FETCH_CONTACTS_SUCCESS,
    data,
    isLoading: false,
    isInvalidated: false,
    status: 200,
  };
};

const errorContact = (error) => {
  return {
    type: FETCH_CONTACT_FAILURE,
    isInvalidated: true,
    isLoading: false,
    status: error.status,
    errors: error.errors
  };
};

export const invalidateContacts = (status) => {
  return {
    type: FETCH_CONTACTS_FAILURE,
    status
  };
};

export const invalidateContact = (status) => {
  return {
    type: FETCH_CONTACT_FAILURE,
    status
  };
};

const fetchContacts = (uuid, offset = request.offset) => (dispatch, getState) => {
  dispatch(requestContacts());
  if (uuid) {
    return api.fetchContacts(uuid)
      .then(res => res.data)
      .then(data => {
        return dispatch(receiveContacts(data, getState()));
      })
      .catch(error => {
        return dispatch(invalidateContacts(error.response.status));
      });
  }
  return api.fetchContacts(null, offset)
    .then(res => res.data)
    .then(data => {
      request.data = request.data.concat(data);
      if (data.length === 200) {
        request.offset += 200;
        return dispatch(fetchContacts(null, request.offset));
      }
      return dispatch(receiveAllContacts(request.data));
    })
    .catch(error => {
      return dispatch(invalidateContacts(error.response.status));
    });
};

const setContacts = (uuid, form) => (dispatch, getState) => {
  dispatch(requestContact());
  return api.setContacts(uuid, form)
    .then(res => {
      if (res.data.errors) {
        return dispatch(errorContact({
          ...res.data,
          status: res.status
        }));
      }
      return dispatch(receiveContact(res.data, getState()));
    }).catch(error => {
      return dispatch(invalidateContact(error.response.status));
    });
};

const initialState = {
  isLoading: false,
  isInvalidated: false,
  data: [],
  status: null,
  fetchedAt: null,
  updatedAt: null
};

export default function(state = initialState, action) {
  switch (action.type) {
  
  case LOGOUT_USER:
    return initialState;
  
  case FETCH_CONTACTS_FAILURE:
    return {
      ...state,
      status: action.status,
      isLoading: action.isLoading,
      isInvalidated: action.isInvalidated,
      errors: action.errors
    };
  
  case FETCH_CONTACTS_REQUEST:
    return {
      ...state,
      isLoading: action.isLoading
    };
  
  case FETCH_CONTACTS_SUCCESS:
    return {
      ...state,
      data: action.data,
      errors: false,
      status: action.status,
      isLoading: action.isLoading,
      isInvalidated: action.isInvalidated,
      fetchedAt: Date.now(),
    };
  
  case FETCH_CONTACT_FAILURE:
    return {
      ...state,
      status: action.status,
      isLoading: action.isLoading,
      isInvalidated: action.isInvalidated,
      errors: action.errors
    };

  case FETCH_CONTACT_REQUEST:
    return {
      ...state,
      isLoading: action.isLoading
    };

  case FETCH_CONTACT_SUCCESS:
    return {
      ...state,
      data: action.data,
      errors: false,
      status: action.status,
      isLoading: action.isLoading,
      isInvalidated: action.isInvalidated,
      updatedAt: Date.now()
    };

  default:
    return state;
  }
};

export {
  initialState,
  fetchContacts,
  setContacts
};
