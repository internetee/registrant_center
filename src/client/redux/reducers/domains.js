import api from '../../utils/api';
import {FETCH_DOMAINS_REQUEST, FETCH_DOMAINS_SUCCESS, FETCH_DOMAINS_FAILURE, LOCK_DOMAIN_REQUEST, LOCK_DOMAIN_SUCCESS, UNLOCK_DOMAIN_REQUEST, UNLOCK_DOMAIN_SUCCESS, LOGOUT_USER} from '../actions';

const request = {
  data: [],
  offset: 0,
};

const requestDomains = () => {
  return {
    type: FETCH_DOMAINS_REQUEST,
    isLoading: true
  };
};

const receiveDomain = (data, state) => {
  return {
    type: FETCH_DOMAINS_SUCCESS,
    status: 200,
    data: state.domains.data.map(item => (item.id === data.id) ? data : item),
    isLoading: false,
    isInvalidated: false,
  };
};

const receiveAllDomains = (data) => {
  return {
    type: FETCH_DOMAINS_SUCCESS,
    status: 200,
    data,
    isLoading: false,
    isInvalidated: false,
  };
};

const requestDomainLock = () => {
  return {
    type: LOCK_DOMAIN_REQUEST,
    isLoading: true
  };
};

const receiveDomainLock = (data, state) => {
  return {
    type: LOCK_DOMAIN_SUCCESS,
    status: 200,
    isLoading: false,
    isInvalidated: false,
    data: state.domains.data.map(item => (item.id === data.id) ? data : item)
  };
};

const requestDomainUnlock = () => {
  return {
    type: UNLOCK_DOMAIN_REQUEST,
    isLoading: true
  };
};

const receiveDomainUnlock = (data, state) => {
  return {
    type: UNLOCK_DOMAIN_SUCCESS,
    status: 200,
    isLoading: false,
    isInvalidated: false,
    data: state.domains.data.map(item => (item.id === data.id) ? data : item)
  };
};

const invalidateRequest = (status) => {
  return {
    type: FETCH_DOMAINS_FAILURE,
    status,
    isLoading: false,
    isInvalidated: true
  };
};

const fetchDomains = (uuid, offset = request.offset) => (dispatch, getState) => {
  dispatch(requestDomains());
  if (uuid) {
    return api.fetchDomains(uuid)
      .then(res => res.data)
      .then(data => {
        return dispatch(receiveDomain(data, getState()));
      })
      .catch(error => {
        return dispatch(invalidateRequest(error.response.status));
      });
  }
  return api.fetchDomains(null, offset)
    .then(res => res.data)
    .then(data => {
      request.data = request.data.concat(data);
      if (data.length === 200) {
        request.offset += 200;
        return dispatch(fetchDomains(null, request.offset));
      }
      return dispatch(receiveAllDomains(request.data));
    })
    .catch(error => {
      dispatch(invalidateRequest(error.response.status));
    });
};

const lockDomain = (uuid) => (dispatch, getState) => {
  dispatch(requestDomainLock());
  return api.setDomainRegistryLock(uuid)
    .then(res => res.data)
    .then(data => {
      return dispatch(receiveDomainLock(data, getState()));
    })
    .catch(error => {
      return dispatch(invalidateRequest(error.response.status));
    });
};

const unlockDomain = (uuid) => (dispatch, getState) => {
  dispatch(requestDomainUnlock());
  return api.deleteDomainRegistryLock(uuid)
    .then(res => res.data)
    .then(data => {
      return dispatch(receiveDomainUnlock(data, getState()));
    })
    .catch(error => {
      return dispatch(invalidateRequest(error.response.status));
    });
};

const initialState = {
  isLoading: false,
  isInvalidated: false,
  data: [],
  status: null,
  fetchedAt: null,
  updatedAt: null
};

export {
  initialState,
  fetchDomains,
  lockDomain,
  unlockDomain
};

export default function(state = initialState, action) {
  switch (action.type) {
  case LOGOUT_USER:
    return initialState;
    
  case FETCH_DOMAINS_FAILURE:
    return {
      ...state,
      status: action.status,
      isLoading: action.isLoading,
      isInvalidated: action.isInvalidated
    };
    
  case FETCH_DOMAINS_REQUEST:
    return {
      ...state,
      isLoading: action.isLoading
    };
  
  case FETCH_DOMAINS_SUCCESS:
    return {
      ...state,
      data: action.data,
      status: action.status,
      isLoading: action.isLoading,
      isInvalidated: action.isInvalidated,
      fetchedAt: Date.now()
    };
  
  case LOCK_DOMAIN_REQUEST:
    return {
      ...state,
      isLoading: action.isLoading
    };
  
  case LOCK_DOMAIN_SUCCESS:
    return {
      ...state,
      status: action.status,
      data: action.data,
      isLoading: action.isLoading,
      isInvalidated: action.isInvalidated,
      updatedAt: Date.now()
    };
  
  case UNLOCK_DOMAIN_REQUEST:
    return {
      ...state,
      isLoading: action.isLoading
    };
  
  case UNLOCK_DOMAIN_SUCCESS:
    return {
      ...state,
      status: action.status,
      data: action.data,
      isLoading: action.isLoading,
      isInvalidated: action.isInvalidated,
      updatedAt: Date.now()
    };
  
  default:
    return state;
  }
};