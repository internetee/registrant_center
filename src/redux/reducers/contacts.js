import api from '../../utils/api';
import {
    DO_UPDATE_CONTACTS_REQUEST,
    DO_UPDATE_CONTACTS_SUCCESS,
    DO_UPDATE_CONTACTS_FAILURE,
    FETCH_CONTACT_REQUEST,
    FETCH_CONTACT_SUCCESS,
    FETCH_CONTACT_FAILURE,
    FETCH_CONTACTS_REQUEST,
    FETCH_CONTACTS_SUCCESS,
    FETCH_CONTACTS_FAILURE,
    UPDATE_CONTACT_REQUEST,
    UPDATE_CONTACT_SUCCESS,
    UPDATE_CONTACT_FAILURE,
    UPDATE_CONTACTS_REQUEST,
    UPDATE_CONTACTS_SUCCESS,
    LOGOUT_USER,
} from '../actions';

let request = {
    data: [],
    offset: 0,
};

const requestContact = () => ({
    type: FETCH_CONTACT_REQUEST,
});

const receiveContact = (payload) => ({
    payload,
    type: FETCH_CONTACT_SUCCESS,
});

const requestContacts = () => ({
    type: FETCH_CONTACTS_REQUEST,
});

const receiveContacts = (payload) => {
    request = {
        data: [],
        offset: 0,
    };
    return {
        payload,
        type: FETCH_CONTACTS_SUCCESS,
    };
};

const requestContactUpdate = () => ({
    type: UPDATE_CONTACT_REQUEST,
});

const receiveContactUpdate = (payload) => ({
    payload,
    type: UPDATE_CONTACT_SUCCESS,
});

const failContactUpdate = (payload) => ({
    payload,
    type: UPDATE_CONTACT_FAILURE,
});

const invalidateContacts = () => ({
    type: FETCH_CONTACTS_FAILURE,
});

const invalidateContact = () => ({
    type: FETCH_CONTACT_FAILURE,
});

const doUpdateContacts = (payload) => ({
    payload,
    type: DO_UPDATE_CONTACTS_REQUEST,
});

const receiveDoUpdateContacts = (payload) => ({
    payload,
    type: DO_UPDATE_CONTACTS_SUCCESS,
});

const failDoUpdateContacts = (payload) => ({
    payload,
    type: DO_UPDATE_CONTACTS_FAILURE,
});

const requestUpdateContacts = (payload) => ({
    payload,
    type: UPDATE_CONTACTS_REQUEST,
});

const updateContacts = (payload) => ({
    payload,
    type: UPDATE_CONTACTS_SUCCESS,
});

const fetchUpdateContacts = (uuid) => (dispatch) => {
    dispatch(doUpdateContacts());
    return api
        .fetchUpdateContacts(uuid, false)
        .then((res) => res.data)
        .then((data) => {
            return dispatch(receiveDoUpdateContacts(data));
        })
        .catch(() => {
            return dispatch(failDoUpdateContacts());
        });;
};

const updateContactsConfirm = (uuid) => (dispatch) => {
    dispatch(requestUpdateContacts());
    return api
        .updateContacts(uuid, false)
        .then((res) => res.data)
        .then((data) => {
            return dispatch(updateContacts(data));
        });
};

const fetchContact = (uuid) => (dispatch) => {
    dispatch(requestContact());
    return api
        .fetchContacts(uuid, false)
        .then((res) => res.data)
        .then((data) => {
            return dispatch(receiveContact(data));
        })
        .catch(() => {
            return dispatch(invalidateContact());
        });
};

const fetchContacts =
    (uuid, offset = request.offset) =>
    (dispatch) => {
        if (uuid) {
            dispatch(requestContact());
            return api
                .fetchContacts(uuid, offset, true)
                .then((res) => res.data)
                .then(async (data) => {
                    return dispatch(receiveContact(data));
                })
                .catch(() => {
                    return dispatch(invalidateContact());
                });
        }
        dispatch(requestContacts());
        return api
            .fetchContacts(null, offset)
            .then((res) => res.data)
            .then((data) => {
                request.data = request.data.concat(data);
                if (data.length === 200) {
                    request.offset += 200;
                    return dispatch(fetchContacts(null, request.offset));
                }
                return dispatch(receiveContacts(request.data));
            })
            .catch(() => {
                return dispatch(invalidateContacts());
            });
    };

const updateContact = (uuid, form) => (dispatch) => {
    dispatch(requestContactUpdate());
    return api
        .updateContact(uuid, form)
        .then((res) => {
            if (res.data.errors) {
                return dispatch(
                    failContactUpdate({
                        ...res.data,
                        code: res.status,
                        type: 'whois',
                    })
                );
            }
            return dispatch(receiveContactUpdate(res.data));
        })
        .catch((error) => {
            return dispatch(
                failContactUpdate({
                    code: error.response.status,
                    type: 'whois',
                })
            );
        });
};

const initialState = {
    data: {},
    ids: [],
    isLoading: false,
    message: null,
};

export default function reducer(state = initialState, { payload, type }) {
    switch (type) {
        case LOGOUT_USER:
            return initialState;

        case DO_UPDATE_CONTACTS_REQUEST:
            return {
                ...state,
                isLoading: true,
            };

        case DO_UPDATE_CONTACTS_SUCCESS:
            return {
                ...state,
                data: {
                    ...state.data,
                },
                isLoading: false,
                message: null,
            };

        case DO_UPDATE_CONTACTS_FAILURE:
            return {
                ...state,
                isLoading: false,
            };

        case UPDATE_CONTACTS_REQUEST:
            return {
                ...state,
                isLoading: true,
            };

        case UPDATE_CONTACTS_SUCCESS:
            return {
                ...state,
                data: {
                    ...state.data,
                },
                isLoading: false,
                message: null,
            };

        case FETCH_CONTACTS_FAILURE:
            return {
                ...state,
                isLoading: false,
            };

        case FETCH_CONTACTS_REQUEST:
            return {
                ...state,
                isLoading: true,
            };

        case FETCH_CONTACTS_SUCCESS:
            return {
                ...state,
                data: payload.reduce(
                    (acc, item) => ({
                        ...acc,
                        [item.id]: item,
                    }),
                    {}
                ),
                ids: payload.map((item) => item.id),
                isLoading: false,
                message: null,
            };

        case FETCH_CONTACT_FAILURE:
            return {
                ...state,
                isLoading: false,
            };

        case FETCH_CONTACT_REQUEST:
            return {
                ...state,
                isLoading: true,
            };

        case FETCH_CONTACT_SUCCESS:
            return {
                ...state,
                data: {
                    ...state.data,
                    [payload.id]: payload,
                },
                ids: state.ids.includes(payload.id) ? state.ids : [...state.ids, payload.id],
                isLoading: false,
                message: null,
            };

        case UPDATE_CONTACT_REQUEST:
            return {
                ...state,
                isLoading: true,
            };

        case UPDATE_CONTACT_SUCCESS:
            return {
                ...state,
                data: { ...state.data, [payload.id]: payload },
                isLoading: false,
                message: {
                    code: 200,
                    type: 'whois',
                },
            };

        case UPDATE_CONTACT_FAILURE:
            return {
                ...state,
                isLoading: false,
                message: payload,
            };

        default:
            return state;
    }
}

export {
    initialState,
    fetchContact,
    fetchContacts,
    updateContact,
    receiveContacts,
    receiveDoUpdateContacts,
    fetchUpdateContacts,
    updateContactsConfirm,
};
