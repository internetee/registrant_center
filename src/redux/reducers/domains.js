/* eslint-disable camelcase */
import api from '../../utils/api';
import {
    FETCH_DOMAIN_REQUEST,
    FETCH_DOMAIN_SUCCESS,
    FETCH_DOMAIN_FAILURE,
    FETCH_DOMAINS_REQUEST,
    FETCH_DOMAINS_SUCCESS,
    FETCH_DOMAINS_FAILURE,
    LOCK_DOMAIN_REQUEST,
    LOCK_DOMAIN_SUCCESS,
    LOCK_DOMAIN_FAILURE,
    UNLOCK_DOMAIN_REQUEST,
    UNLOCK_DOMAIN_SUCCESS,
    UNLOCK_DOMAIN_FAILURE,
    LOGOUT_USER,
} from '../actions';
import { fetchContacts } from './contacts';
import domainStatuses from '../../utils/domainStatuses.json';

let request = {
    data: {
        count: 0,
        domains: [],
    },
    offset: 0,
};

const parseDomain = (domain, shouldFetchContacts = false, simplify = false) => {
    const { admin_contacts, registrant, tech_contacts } = domain;
    const statuses = domain.statuses.sort(
        (a, b) => domainStatuses[a].priority - domainStatuses[b].priority
    );

    const contacts = (simplify
        ? [
              ...(registrant && [
                  {
                      ...registrant,
                      roles: ['registrant'],
                  },
              ]),
          ]
        : [
              ...(admin_contacts && admin_contacts.map((item) => ({ ...item, roles: ['admin'] }))),
              ...(tech_contacts && tech_contacts.map((item) => ({ ...item, roles: ['tech'] }))),
              ...(registrant && [
                  {
                      ...registrant,
                      roles: ['registrant'],
                  },
              ]),
          ]
    ).flat();

    return {
        ...domain,
        contacts: contacts.reduce(
            (acc, { id, roles, ...rest }) => ({
                ...acc,
                [id]: {
                    ...rest,
                    id,
                    roles: acc[id] ? [...acc[id].roles, ...roles] : roles,
                },
            }),
            {}
        ),
        isLockable: ['pendingDelete', 'serverHold'].every((status) => !statuses.includes(status)),
        isLocked:
            domain.locked_by_registrant_at &&
            [
                'serverUpdateProhibited',
                'serverDeleteProhibited',
                'serverTransferProhibited',
            ].every((r) => statuses.includes(r)),
        shouldFetchContacts,
    };
};

const requestDomain = () => ({
    type: FETCH_DOMAIN_REQUEST,
});

const receiveDomain = (payload) => ({
    payload,
    type: FETCH_DOMAIN_SUCCESS,
});

const failDomain = () => ({
    type: FETCH_DOMAIN_FAILURE,
});

const requestDomains = () => ({
    type: FETCH_DOMAINS_REQUEST,
});

const receiveDomains = (payload, simplify = false) => {
    request = {
        data: {
            count: 0,
            domains: [],
        },
        offset: 0,
    };
    return {
        payload,
        simplify,
        type: FETCH_DOMAINS_SUCCESS,
    };
};

const failDomains = () => ({
    type: FETCH_DOMAINS_FAILURE,
});

const requestDomainLock = () => ({
    type: LOCK_DOMAIN_REQUEST,
});

const receiveDomainLock = (payload) => ({
    payload,
    type: LOCK_DOMAIN_SUCCESS,
});

const failDomainLock = (payload) => ({
    payload,
    type: LOCK_DOMAIN_FAILURE,
});

const requestDomainUnlock = () => ({
    type: UNLOCK_DOMAIN_REQUEST,
});

const receiveDomainUnlock = (payload) => ({
    payload,
    type: UNLOCK_DOMAIN_SUCCESS,
});

const failDomainUnlock = (payload) => ({
    payload,
    type: UNLOCK_DOMAIN_FAILURE,
});

const fetchDomain = (uuid) => (dispatch) => {
    dispatch(requestDomain());
    return api
        .fetchDomains(uuid)
        .then((res) => res.data)
        .then(async (data) => {
            const domain = parseDomain(data);
            await Promise.all(
                Object.keys(domain.contacts).map((id) => dispatch(fetchContacts(id)))
            );
            return dispatch(receiveDomain(domain));
        })
        .catch(() => {
            return dispatch(failDomain());
        });
};

const fetchDomains = (offset = request.offset, simplify = false) => (dispatch) => {
    dispatch(requestDomains());
    return api
        .fetchDomains(null, offset, simplify)
        .then((res) => res.data)
        .then((data) => {
            request.data.domains = request.data.domains.concat(data.domains);
            request.data.count = data.count;
            if (request.data.domains.length !== data.count) {
                request.offset += 200;
                return dispatch(fetchDomains(request.offset, simplify));
            }
            return dispatch(receiveDomains(request.data, simplify));
        })
        .catch(() => {
            dispatch(failDomains());
        });
};

const lockDomain = (uuid) => (dispatch) => {
    dispatch(requestDomainLock());
    return api
        .setDomainRegistryLock(uuid)
        .then((res) => res.data)
        .then((data) => {
            return dispatch(receiveDomainLock(data));
        })
        .catch((error) => {
            return dispatch(
                failDomainLock({
                    code: error.response.status,
                })
            );
        });
};

const unlockDomain = (uuid) => (dispatch) => {
    dispatch(requestDomainUnlock());
    return api
        .deleteDomainRegistryLock(uuid)
        .then((res) => res.data)
        .then((data) => {
            return dispatch(receiveDomainUnlock(data));
        })
        .catch((error) => {
            return dispatch(
                failDomainUnlock({
                    code: error.response.status,
                })
            );
        });
};

const initialState = {
    data: {
        count: 0,
        domains: [],
    },
    ids: [],
    isLoading: false,
    message: null,
};

export default function reducer(state = initialState, { payload, type, simplify = false }) {
    switch (type) {
        case LOGOUT_USER:
            return initialState;

        case FETCH_DOMAIN_REQUEST:
            return {
                ...state,
                isLoading: true,
            };

        case FETCH_DOMAIN_SUCCESS:
            return {
                data: { ...state.data, [payload.id]: payload },
                ids: state.ids.includes(payload.id) ? state.ids : [...state.ids, payload.id],
                isLoading: false,
            };

        case FETCH_DOMAIN_FAILURE:
            return {
                ...state,
                isLoading: false,
            };

        case FETCH_DOMAINS_REQUEST:
            return {
                ...state,
                isLoading: true,
            };

        case FETCH_DOMAINS_SUCCESS:
            return {
                ...state,
                data: {
                    count: payload.count,
                    domains: payload.domains.reduce(
                        (acc, item) => ({
                            ...acc,
                            [item.id]: parseDomain(item, true, simplify),
                        }),
                        {}
                    ),
                },
                ids: payload.domains.map((item) => item.id),
                isLoading: false,
            };

        case FETCH_DOMAINS_FAILURE:
            return {
                ...state,
                isLoading: false,
            };

        case LOCK_DOMAIN_REQUEST:
            return {
                ...state,
            };

        case LOCK_DOMAIN_SUCCESS:
            return {
                ...state,
                data: {
                    ...state.data,
                    [payload.id]: parseDomain(payload),
                },
                message: {
                    code: 200,
                    type: 'domainLock',
                },
            };

        case LOCK_DOMAIN_FAILURE:
            return {
                ...state,
                message: {
                    ...payload,
                    type: 'domainLock',
                },
            };

        case UNLOCK_DOMAIN_REQUEST:
            return {
                ...state,
            };

        case UNLOCK_DOMAIN_SUCCESS:
            return {
                ...state,
                data: {
                    ...state.data,
                    [payload.id]: parseDomain(payload),
                },
                message: {
                    code: 200,
                    type: 'domainUnlock',
                },
            };

        case UNLOCK_DOMAIN_FAILURE:
            return {
                ...state,
                message: {
                    ...payload,
                    type: 'domainUnlock',
                },
            };

        default:
            return state;
    }
}

export { initialState, fetchDomain, fetchDomains, lockDomain, parseDomain, unlockDomain };
