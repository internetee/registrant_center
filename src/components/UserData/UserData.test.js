import React from 'react';
import 'core-js';
import { shallow } from 'enzyme';
import UserData from './UserData';
import Providers from '../../__mocks__/Providers';
import contacts from '../../__mocks__/contacts';
import domains from '../../__mocks__/domains';

describe('components/UserData', () => {
	it('should show content', () => {
		const cmp = shallow(
			<Providers>
				<UserData contacts={contacts} domains={domains} />
			</Providers>
		);
		expect(cmp).toMatchSnapshot();
	});
});
