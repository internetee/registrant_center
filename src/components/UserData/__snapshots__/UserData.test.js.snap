// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/UserData should show content 1`] = `ShallowWrapper {}`;
